test/e2e/revoke_credentials_test.go

```diff
@@ -0,0 +1,98 @@
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package e2e
+
+import (
+	"context"
+	"time"
+
+	. "github.com/onsi/ginkgo/v2"
+	. "github.com/onsi/gomega"
+	"k8s.io/apimachinery/pkg/util/rand"
+
+	"github.com/Azure/ARO-HCP/test/util/framework"
+	"github.com/Azure/ARO-HCP/test/util/labels"
+)
+
+var _ = Describe("Revoke Credentials", func() {
+	Context("Positive Credential Revocation", func() {
+		It("successfully revokes cluster credentials",
+			labels.RequireNothing, labels.High, labels.Positive,
+			func(ctx context.Context) {
+				clusterName := "revoke-creds-cluster-" + rand.String(6)
+				tc := framework.NewTestContext()
+
+				By("creating resource group and cluster for credential revocation")
+				resourceGroup, err := tc.NewResourceGroup(ctx, "revoke-credentials-test", "uksouth")
+				Expect(err).NotTo(HaveOccurred())
+
+				By("deploying cluster for credential revocation testing")
+				_, err = framework.CreateBicepTemplateAndWait(ctx,
+					tc.GetARMResourcesClientFactoryOrDie(ctx).NewDeploymentsClient(),
+					*resourceGroup.Name,
+					"revoke-creds-cluster",
+					framework.Must(TestArtifactsFS.ReadFile("test-artifacts/generated-test-artifacts/demo.json")),
+					map[string]interface{}{
+						"clusterName": clusterName,
+					},
+					45*time.Minute,
+				)
+				Expect(err).NotTo(HaveOccurred())
+
+				By("verifying cluster exists before credential operations")
+				clusterClient := tc.Get20240610ClientFactoryOrDie(ctx).NewHcpOpenShiftClustersClient()
+				cluster, err := clusterClient.Get(ctx, *resourceGroup.Name, clusterName, nil)
+				Expect(err).NotTo(HaveOccurred())
+				Expect(cluster.HcpOpenShiftCluster.ID).NotTo(BeNil())
+
+				By("first requesting admin credentials for the cluster")
+				adminRESTConfig, err := framework.GetAdminRESTConfigForHCPCluster(
+					ctx,
+					clusterClient,
+					*resourceGroup.Name,
+					clusterName,
+					10*time.Minute,
+				)
+				Expect(err).NotTo(HaveOccurred())
+				Expect(adminRESTConfig).NotTo(BeNil())
+
+				By("verifying initial admin credentials work")
+				Expect(framework.VerifyHCPCluster(ctx, adminRESTConfig)).To(Succeed())
+
+				By("revoking the cluster credentials via ARM API")
+				poller, err := clusterClient.BeginRevokeCredentials(ctx, *resourceGroup.Name, clusterName, nil)
+				Expect(err).NotTo(HaveOccurred())
+
+				By("waiting for revocation operation to complete")
+				_, err = poller.PollUntilDone(ctx, nil)
+				Expect(err).NotTo(HaveOccurred())
+
```

Comment:
shouldn't we verify our credentials no longer work here? Subsequent calls with our kubeconfig should get us a 403, right?

Reply:
another missing test case that's worth a test: revoke credentials is expected to cause _all_ customer created credentials to fail as above. Test case would be:

1. Create cred A, GET on default Service returns 2xx
2. Create cred B, GET on default Service returns 2xx
3. Revoke
4. Cred A GET on default Service returns 403
5. Cred B GET on default Service returns 403

To improve the test further, we could create an arbitrarily long list of credentials, revoke, and test each of them, so it's easy for us to test scalability concerns as well (e.g. 10 or 100 creds would be easy to scale the test to)

Reply:
Added. Thanks! Currently, we are using `credentialCount := 3`

Reply:
Also added CONSULTANT HYPOTHESIS TEST